test/manifold/deferred_test.clj

```diff
@@ -118,7 +118,57 @@
         (is (= ["cat" "cat" "cat"]
                @(d/let-flow [a d
                              b (do a *test-dynamic-var*)]
-                  [a b *test-dynamic-var*])))))))
+                  [a b *test-dynamic-var*]))))))
+
+  (let [start          (System/currentTimeMillis)
+        future-timeout (d/future (Thread/sleep 500) "b")
+        expected       (d/future (Thread/sleep 5) "cat")]
+    @(d/let-flow [x (d/alt future-timeout expected)]
+       x)
+
+    (is (>= 300 (- (System/currentTimeMillis) start))
+        "Alt in let-flow should only take as long as the first deferred to finish."))
+
+  (is (every? #(= "cat" %)
+              (for [i (range 50)]
+                (let [future-timeout (d/future (Thread/sleep 100) "b")
+                      expected       (d/future (Thread/sleep 5) "cat")]
+                  @(d/let-flow [x (d/alt future-timeout expected)]
+                     x))))
+      "Resolution of deferreds in alt inside a let-flow should always be consistent.")
+
+  (let [start          (System/currentTimeMillis)
+        future-timeout (d/future (Thread/sleep 300) "b")
+        expected       (d/future (Thread/sleep 5) "cat")]
+    (is (= "cat"
+           @(d/let-flow [x (d/alt future-timeout expected)
+                         y (d/alt x future-timeout)]
+              (d/alt future-timeout y)))
+        "Alts referencing newly introduced symbols shouldn't cause compiler errors.")
+    (is (>= 200 (- (System/currentTimeMillis) start))
+        "Alt in body should only take as long as the first deferred to finish."))
```

Comment:
I don't think we can make this change for let-flow bodies, only bindings. The docstring explicitly says "The body will only be executed once all of the let-bound
   values, even ones only used for side effects, have been computed.", and some users may be relying on that.

More generally, an alt in the body only makes sense if some of its arguments *aren't* bound in the let-flow. If all of them are, then it's essentially random, which is fine, but may not be what they expect.